torch/csrc/jit/tensorexpr/stmt.h

```diff
@@ -112,6 +112,14 @@ class Block : public StmtNode<Block> {
     return stmts_.size();
   }
 
+  void prepend_stmt(Stmt* s) {
+    if (s->get_parent()) {
```

Comment:
Why does this have to be a root? Also the error message here might be difficult to interpret.

Reply:
This is how we're protecting against using statements in several other statements. When one inserts a statement into a block, the inserted statement must be without parent - it either needs to be created from scratch, or created by `Stmt::clone`. If you take a statement from existing block and try to insert it into another block, it will correctly fail at this check.

Also, please note that this PR does not introduce this mechanism, but rather follows already existing one (see `append_stmt`).